### PR TITLE
Interp

### PR DESCRIPTION
**Pull request by sofiachiarenza**

### CODE REVIEW

File: README.md
Comment (marcobonici) [inline README.md:1]: Please, add a bit more info.
1. What the code is supposed to do.
2. How the code work (a small snippet).
3. Authors and everything.

REgarding point 2, it does not have to be self consistent: you might feed some `Pk` and `windows` even without showing how to evaluate them.
You can get insptied by `Capse` repo or chatgpt.

File: src/chebcoefs.jl [hunk 1]
```diff
@@ -1,24 +1,27 @@
+# FUNCTIONS FOR THE DECOMPOSITION OF THE POWER SPECTRUM ON THE BASIS OF THE CHEBYSHEV POLYNOMIALS, CENTRAL PART OF BLAST
 """
-    plan_fft(vals::AbstractArray{<:Number,N})
+    plan_fft(vals::AbstractArray{<:Number, N}, axis::Int)
 
-Create an FFTW real-to-real (R2R) transformation plan for the first axis of a given multidimensional array `vals`. 
-In practise, the `vals` array is the power spectrum P(k,χ). The first axis should then contains the wavenumbers `k`, while the second axis contains the `χ` information.
+Create an FFTW real-to-real (R2R) transformation plan for a specified axis of a given multidimensional array `vals`. 
+In practice, the `vals` array is often the power spectrum P(k,χ): the specified axis should contain the wavenumbers `k`. So if the power spectrum is given in a matrix of shape (nk, nχ), axis should be `1`. Instead, axis = `2` should be used for a matrix of shape (nχ, nk).
```
Comment (marcobonici) [inline src/chebcoefs.jl]: I am confused. The chosen axis should depend only  on which axis you want to perform the FFT on, right? Sometimes it is the axis related to `z`, sometimes the axis realted to `chi`. Right?

File: src/chebcoefs.jl [hunk 2]
```diff
@@ -43,3 +46,92 @@ function fast_chebcoefs(vals::AbstractArray, plan::FFTW.r2rFFTWPlan)
...
+    coefs = fast_chebcoefs(pk, plan)
+    new_χs = make_grid(bg, R)
+    x = resample_redshifts(bg, grid, new_χs)
+    chebyshevs = chebyshev_polynomials(x, length(z_nodes), minimum(x), maximum(x))
+    pk_interp = coefs' * chebyshevs  #TODO: understand how to handle pk sizes
+    return reshape(pk_interp, size(pk,2),  length(bg.χz_array), length(R))
+end
+
+"""
+    correlated_power_spectrum(pk::AbstractArray{T,3}) where T
```
Comment (marcobonici) [inline src/chebcoefs.jl]: I don't like the name. Why correlated? I would call it the "unequal_time_power_spectrum" or something similar...but I am open to discussing it :)

**Review by marcobonici — CHANGES_REQUESTED** [review]
Hi @sofiachiarenza , I gave some comments. Mostly minor stuff.

**Comment by sofiachiarenza** [comment]
> Hi @sofiachiarenza , I gave some comments. Mostly minor stuff.

Done! ;)

**Comment by marcobonici** [comment]
Can you just update, in the Readme, the couple of places without the Unicode characters?

\Tilde{T}^{AB}_\ell(\chi_1,\chi_2), for instance.

**Review by marcobonici — APPROVED** [review]

**Pull request merged**